x/hard/keeper/borrow_test.go

```diff
@@ -561,4 +811,37 @@ func (suite *KeeperTestSuite) TestValidateBorrow() {
 		sdk.NewCoins(sdk.NewCoin("ukava", availableToBorrow.AmountOf("ukava"))),
 	)
 	suite.Require().NoError(err)
+
+	// now that it's all that you can borrow, we shouldn't be able to borrow anything
+	err = suite.keeper.Borrow(
+		suite.ctx,
+		borrower,
+		sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(2*USDX_CF))),
+	)
+	suite.Require().Error(err)
+	suite.Require().True(strings.Contains(err.Error(), "available to borrow: exceeds module account balance"))
+
+	suite.ctx = suite.ctx.WithBlockTime(suite.ctx.BlockTime().Add(blockDuration))
+	hard.BeginBlocker(suite.ctx, suite.keeper)
+
+	// Should error since ukava has become insolvent at this point
+	err = suite.keeper.Borrow(
+		suite.ctx,
+		borrower,
+		sdk.NewCoins(sdk.NewCoin("ukava", availableToBorrow.AmountOf("ukava"))),
+	)
+	suite.Require().Error(err)
+	suite.Require().True(strings.Contains(err.Error(), "protocol reserves exceed available cash"))
```

Comment:
not blocking merge, can be updated in the next pr if you want - just easier error checking

```suggestion
	suite.Require().ErrorContains(err, "protocol reserves exceed available cash")
```